src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -334,23 +334,37 @@ case class Lambda(posn: Position, param: String, body: AST) extends AST(posn, bo
 case class Apply(posn: Position, fn: String, args: Array[AST]) extends AST(posn, args) {
   override def typecheckThis(): Type = {
     (fn, args) match {
+      case ("str", rhs) =>
+        if (rhs.length != 1)
+          parseError("str expects 1 argument")
+        if (!rhs.head.`type`.isRealizable)
+          parseError("Argument to str of unrealizable type")
```

Comment:
Previously, we included the offending type in the message. I think we should continue to do so.